ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs

```diff
@@ -903,6 +904,7 @@ withPeerStateActions PeerStateActionsArguments {
       case res of
         Nothing -> do
           -- timeout fired
+          Mux.stopMux pchMux
           _ <- unregisterOutboundConnection spsConnectionManager (remoteAddress pchConnectionId)
```

Comment:
You can leave remove this call. There's no need to unregister a connection that died.

Reply:
I left it there to match https://github.com/input-output-hk/ouroboros-network/blob/f2425ffba98a60c253d6ba136b5af7006b40bd85/ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs#L834-L837 .

But will change both places to not call `unregisterOutboundConnection` after `stopMux`.

Reply:
You can remove both in a separate commit.  Both are complex `no-op`: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs?plain=1#L2035